src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t create country " + country, e);
+        }
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Country country = null;
+        try (Session session = factory.openSession()) {
+            country = session.get(Country.class, id);
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get country by Id " + id, e);
+        }
+        return Optional.ofNullable(country);
```

Comment:
I think it would be better to put return in the try block and also initialize the country there